File: 5243-145-1207-unique-number-of-occurrences/5243-145-1207-unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        HashMap<Integer,Integer> map = new HashMap<>();
        HashSet<Integer> set = new HashSet<>();

        for(int num: arr)
            map.put(num,map.getOrDefault(num,0)+1);

        for(int value: map.values()){
            if(set.contains(value))
                return false;
            set.add(value);
        }
        return true;
    }
}

File: 5243-145-1207-unique-number-of-occurrences/5243-145-1207-unique-number-of-occurrences.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var uniqueOccurrences = function(arr) {
    map = {}
    for(let num of arr)
        map[num] = (map[num] || 0) + 1
    
    set = new Set()

    for(let value of Object.values(map)){
        if(set.has(value))
            return false
        set.add(value)
    }
    return true
};